src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -461,14 +465,57 @@ fn all_constructors<'a, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
                 .map(|v| Variant(v.did))
                 .collect()
         }
+        ty::TyChar if exhaustive_integer_patterns => {
+            let (min, max) = (0u128, char::MAX as u128);
+            value_constructors = true;
+            vec![ConstantRange(ty::Const::from_bits(cx.tcx, min, cx.tcx.types.char),
+                               ty::Const::from_bits(cx.tcx, max, cx.tcx.types.char),
+                               RangeEnd::Included)]
```

Comment:
Hopefully the `char` request is just to initialize this as a two-element vec instead...